plugin/hook/adapter_test.go

```diff
@@ -0,0 +1,622 @@
+				Name:  "",
+				Value: "",
+			},
+		})
+		expected := &pb.HookConfigs{
+			Configs: []*pb.HookConfigs_Config{
+				{
+					Name:  "",
+					Value: "",
+				},
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for non-empty input", func(t *testing.T) {
+		actual := adaptConfigsToProto(models.HookPluginConfigs{
+			models.HookPluginConfig{
+				Name:  "nil",
+				Value: "nil",
+			},
+		})
+		expected := &pb.HookConfigs{
+			Configs: []*pb.HookConfigs_Config{
+				{
+					Name:  "nil",
+					Value: "nil",
+				},
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for partial config parameters", func(t *testing.T) {
+		actual := adaptConfigsToProto(models.HookPluginConfigs{
+			models.HookPluginConfig{
+				Name: "key",
+			},
+		})
+		expected := &pb.HookConfigs{
+			Configs: []*pb.HookConfigs_Config{
+				{
+					Name: "key",
+				},
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for partial config parameters", func(t *testing.T) {
+		actual := adaptConfigsToProto(models.HookPluginConfigs{
+			models.HookPluginConfig{
+				Value: "sec",
+			},
+		})
+		expected := &pb.HookConfigs{
+			Configs: []*pb.HookConfigs_Config{
+				{
+					Value: "sec",
+				},
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should config to proto", func(t *testing.T) {
+		actual := adaptConfigsToProto(models.HookPluginConfigs{
+			models.HookPluginConfig{
+				Name:  "key",
+				Value: "sec",
+			},
+		})
+		expected := &pb.HookConfigs{
+			Configs: []*pb.HookConfigs_Config{
+				{
+					Name:  "key",
+					Value: "sec",
+				},
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+}
+
+func Test_adaptConfigsFromProto(t *testing.T) {
+	t.Run("should work for empty input", func(t *testing.T) {
+		actual := adaptConfigFromProto(nil)
+		expected := models.HookPluginConfigs{}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for empty input", func(t *testing.T) {
+		actual := adaptConfigFromProto(&pb.HookConfigs{
+			Configs: []*pb.HookConfigs_Config{},
+		})
+		expected := models.HookPluginConfigs{}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for non empty inputs", func(t *testing.T) {
+		actual := adaptConfigFromProto(&pb.HookConfigs{
+			Configs: []*pb.HookConfigs_Config{
+				{
+					Name:  "nil",
+					Value: "nil",
+				},
+			},
+		})
+		expected := models.HookPluginConfigs{
+			models.HookPluginConfig{
+				Name:  "nil",
+				Value: "nil",
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("partial config should be adapted from proto correctly", func(t *testing.T) {
+		actual := adaptConfigFromProto(&pb.HookConfigs{
+			Configs: []*pb.HookConfigs_Config{
+				{
+					Name: "key",
+				},
+			},
+		})
+		expected := models.HookPluginConfigs{
+			models.HookPluginConfig{
+				Name: "key",
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("partial config should be adapted from proto correctly", func(t *testing.T) {
```

Comment:
this test and the test above this can be merged into one, there are a lot of test cases that can be merged into one instead of writing them separately. Kindly merge them wherever possible.

Reply:
Is checking only 1 parameter here will cover the case ?